### PR TITLE
[SM-501] Add support for revoking access tokens

### PR DESCRIPTION
**Pull request by Hinton**
## Type of change

```
- [ ] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Adds support for revoking access tokens.

## Code changes

* **src/Api/SecretsManager/Controllers/ServiceAccountsController.cs:** New endpoint.
* **src/Infrastructure.Dapper/SecretsManager/Repositories/ApiKeyRepository.cs**: Support for deleting many api keys.
* **test/Api.IntegrationTest/SecretsManager/Controllers/ServiceAccountsControllerTests.cs**: Tests.

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: src/Api/SecretsManager/Controllers/ServiceAccountsController.cs [hunk 1]
```diff
@@ -129,4 +129,41 @@ public async Task<ListResponseModel<AccessTokenResponseModel>> GetAccessTokens([
...
+            _ => false,
+        };
+
+        if (!hasAccess)
+        {
+            throw new NotFoundException();
+        }
+
+        var accessTokens = await _apiKeyRepository.GetManyByServiceAccountIdAsync(id);
+
```
Comment (Thomas-Avery) [inline src/Api/SecretsManager/Controllers/ServiceAccountsController.cs:164]: Thoughts on separating out this logic into a command? `RevokeAccessTokenCommand`

**Review by Thomas-Avery — COMMENTED** [review]
Do we need to run EF migrations?

**Comment by Hinton** [comment]
We haven't changed any database structure so we shouldn't need any EF migrations.

File: src/Api/SecretsManager/Controllers/ServiceAccountsController.cs [hunk 2]
```diff
@@ -129,4 +133,37 @@ public async Task<ListResponseModel<AccessTokenResponseModel>> GetAccessTokens([
...
+            throw new NotFoundException();
+        }
+
+        var orgAdmin = await _currentContext.OrganizationAdmin(serviceAccount.OrganizationId);
+        var accessClient = AccessClientHelper.ToAccessClient(_currentContext.ClientType, orgAdmin);
+
+        var hasAccess = accessClient switch
+        {
+            AccessClientType.NoAccessCheck => true,
+            AccessClientType.User => await _serviceAccountRepository.UserHasReadAccessToServiceAccount(id, userId),
```
[the remark this answers is not in the document]
Comment (Hinton) [reply src/Api/SecretsManager/Controllers/ServiceAccountsController.cs:158]: It should be. Nice that one of us is awake.

**Review by Thomas-Avery — APPROVED** [review]
LGTM

**Pull request merged**